velox/expression/tests/ExprTest.cpp

```diff
@@ -939,6 +939,40 @@ TEST_F(ExprTest, selectiveLazyLoadingOr) {
   assertEqualVectors(expected, result);
 }
 
+TEST_F(ExprTest, lazyVectorAccessTwiceWithDifferentRows) {
```

Comment:
Currently both these tests only verify that changes local to a single expression tree are working correctly.
Can you also add a test that exercises the changes at ExprSet::eval(). checkout 'evaluateMultiple()' to help you run separate expression trees under a single ExprSet